snappy/overlord.go

```diff
@@ -246,6 +243,80 @@ func ActivateSnap(s *Snap, inter interacter) error {
 	return os.Symlink(filepath.Base(s.basedir), currentDataSymlink)
 }
 
+func UndoUpdateCurrentSymlink(oldSnap, newSnap *Snap, inter interacter) error {
+	if err := removeCurrentSymlink(newSnap, inter); err != nil {
+		return err
+	}
+	return UpdateCurrentSymlink(oldSnap, inter)
+}
+
+func removeCurrentSymlink(s *Snap, inter interacter) error {
+	var err1, err2 error
+
+	// the snap "current" symlink
+	currentActiveSymlink := filepath.Join(s.basedir, "..", "current")
+	err1 = os.Remove(currentActiveSymlink)
+	if err1 != nil && !os.IsNotExist(err1) {
+		logger.Noticef("Failed to remove %q: %v", currentActiveSymlink, err1)
+	} else {
+		err1 = nil
+	}
+
+	// the data "current" symlink
+	currentDataSymlink := filepath.Join(dirs.SnapDataDir, s.Name(), "current")
+	err2 = os.Remove(currentDataSymlink)
+	if err2 != nil && !os.IsNotExist(err2) {
+		logger.Noticef("Failed to remove %q: %v", currentDataSymlink, err2)
+	} else {
+		err2 = nil
+	}
+
+	if err1 != nil && err2 != nil {
+		return fmt.Errorf("removeCurrentSymlink failed with: %v and %v", err1, err2)
+	} else if err1 != nil {
+		return fmt.Errorf("removeCurrentSymlink failed with: %v", err1)
+	} else if err2 != nil {
+		return fmt.Errorf("removeCurrentSymlink failed with: %v", err2)
```

Comment:
Sorry for missing that earlier, but this is an error that may pop up to the user, so I'd suggest something like:

`cannot remove snap current symlink: ...`